### PR TITLE
Migrate http client to Symfony, update cache system accordingly, for PrestaShop 9 only

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Migrate http client to Symfony, update cache system accordingly, for PrestaShop 9 only
| Type?             | new feature
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | ~
| Sponsor company   | ~
| How to test?      | ~

### CODE REVIEW

**Review by boherm — APPROVED** [review]

File: src/Middleware/CachedResponse.php
Comment (matthieu-rolland) [inline src/Middleware/CachedResponse.php:53]: that's nitpicking but I think this code could be shortened a lot by making an array with all the default "info" values, and then doing an `array_merge` with $info to apply the content of `$response->getInfo()`

there must be a way to do the same with an object with `array_map`

**Review by matthieu-rolland — APPROVED** [review]

File: src/Middleware/CachedResponse.php
Comment (matthieu-rolland) [reply src/Middleware/CachedResponse.php:53]: (you actually already have the array with default values in the else condition)

Comment (jolelievre) [reply src/Middleware/CachedResponse.php:53]: Yeah this is a bit complex and mostly to please PHPSTan actually 😅

Comment (matthieu-rolland) [reply src/Middleware/CachedResponse.php:53]: damn, alright ! what wouldn't we do for phpstan :D

**Review by matthieu-rolland — APPROVED** [review]

**Review by tleon — APPROVED** [review]

File: src/Middleware/CachedResponse.php
Comment (jolelievre) [reply src/Middleware/CachedResponse.php:53]: there was probably a better way to solve this, but I admit I was in a rush
and in case the response really return an object at least we're covered

**Comment by jolelievre** [comment]
QA validated via this PR https://github.com/PrestaShop/PrestaShop/pull/37510
In this run https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/12073041885

**Review by boherm — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
